Updating component spec

ajmell7 opened the pull request.

ajmell7 committed:
Update component specification document to combine plots with text

ajmell7 committed:
Fix formatting of bullet points

ajmell7 committed:
Fix newline issue for component specification

ajmell7 committed:
Fix indenting

ajmell7 committed:
Fix capitalization

marissafujimoto commented on docs/component_specification.md:
```diff
@@ -1,6 +1,49 @@
+### <ins>**Embedding Model:**</ins>
+#### **What it does:**
+* Determines how to plot the cancer data into a map of cancer transcriptomics by translating the embedding data frame it is provided into coordinates for the embedding plotter
+#### **Inputs:**
+* Data from the embedding dataframe
+#### **Outputs:**
+* Data mapped to coordinates for the embedding plotter to plot
+#### **Assumptions:**
+* These is an embedding data frame available to pull data from
```
I think the embedding model's input is some form of the raw gene expression values and the output is embedding coordinates.

For the output, I think it's best if components don't explicitly know about how their output is used downstream (e.g. don't mention embedding plotter)

I'm not sure if this is an issue for this level of documentation, but I think we might want to be more specific about the dataframes we are passing. What columns are in the dataframe might be useful.

marissafujimoto commented on docs/component_specification.md:
```diff
@@ -1,6 +1,49 @@
+### <ins>**Filtering Panel:**</ins>
+#### **What it does:**
+* Displays filtering options and takes user input for how to filter the data that is displayed by the embedding plotter
+#### **Inputs:**
+* Age, numeric
+* Gender, string
+* Race, string
```
I think the input here is more abstract, but these are good examples. We really could filter on any kind of categorical or numerical metadata columns.

marissafujimoto reviewed (changes requested):
Looks good so far! I'm just hoping to clarify some of the stuff about input and output a bit more. I think also it might be useful to highlight when an output involves a UI change happening or maybe calling UI changes side effects. If not for this document, just mentally understanding that will be important. Also I'm learning new things you can do with markdown, wow!

ajmell7 committed:
Update to clarify inputs and outputs and add line about UI

ajmell7 committed:
Move UI change from filtering panel to view controller

ajmell7 commented:
Updated the inputs and outputs and added a line about the UI to the view controller in the additional components section. Should anything else affect the UI?

marissafujimoto commented:
> Updated the inputs and outputs and added a line about the UI to the view controller in the additional components section. Should anything else affect the UI?

I think the way I'm imagining the filtering panel it might have a UI? It's something I think we'll figure out later though so no worries.

marissafujimoto reviewed (approved):
Looks good to me!
